src/conversations/Conversations.ts

```diff
@@ -152,6 +160,38 @@ export default class Conversations<ContentTypes = any> {
     return this.decodeInvites(envelopes)
   }
 
+  private async validateConsentSignature(
+    signature: `0x${string}`,
+    timestamp: number,
+    peerAddress: string
+  ): Promise<boolean> {
+    const signatureData = splitSignature(signature)
+    const message = WalletSigner.consentProofRequestText(peerAddress, timestamp)
+    const digest = hexToBytes(hashMessage(message))
+    // Recover public key
+    const publicKey = ecdsaSignerKey(digest, signatureData)
+    return publicKey?.getEthereumAddress() === this.client.address
+  }
+
+  private async handleConsentProof(
+    consentProof: invitation.ConsentProofPayload,
+    peerAddress: string
+  ): Promise<void> {
+    const { signature, timestamp } = consentProof
+    const isValid = await this.validateConsentSignature(
+      signature as `0x${string}`,
+      Number(timestamp),
+      peerAddress
+    )
+    if (!isValid) {
+      return
+    }
+    const consentState = await this.client.contacts.consentState(peerAddress)
```

Comment:
Are we sure that this state will always be current? `decodeInvites` can be run many times if the client is using in-memory caching (common with bots), so I wouldn't want to have repeated consent entries come out of this if you call `conversations.list()` several times with different client instances that don't share a cache.